splice_si_api/src/main/java/com/splicemachine/si/impl/store/CompletedTxnCacheSupplier.java

```diff
@@ -34,72 +31,84 @@
  *         Date: 6/18/14
  */
 public class CompletedTxnCacheSupplier implements TxnSupplier{
-    private ConcurrentLinkedHashMap<Long, TxnView> cache; // autobox for now
+    private TxnView[] cache;
     private final TxnSupplier delegate;
-    private final AtomicLong hits=new AtomicLong();
-    private final AtomicLong requests=new AtomicLong();
-
-    public CompletedTxnCacheSupplier(TxnSupplier delegate,int maxSize,int concurrencyLevel){
-        cache=new ConcurrentLinkedHashMap.Builder<Long, TxnView>()
-                .maximumWeightedCapacity(maxSize)
-                .concurrencyLevel(concurrencyLevel)
-                .build();
-        this.delegate=delegate;
+
+    public CompletedTxnCacheSupplier(TxnSupplier delegate, int maxSize, int concurrencyLevel) {
+        cache = new TxnView[Integer.highestOneBit(maxSize)];    // ensure power of 2
+        this.delegate = delegate;
+    }
+
+    private int idx(long val) {
+        int hashCode = Long.hashCode(val);
+        hashCode ^= (hashCode >>> 20) ^ (hashCode >>> 12);
```

Comment:
Did you get hashing strategy from somewhere or did you come up with it?

Reply:
We use the same hash function in _LongStripedSynchronizer_.
_idx2()_ is my extension to break collisions in _idx()_.

Reply:
I'd rather have the logic in a single place (some utils class?) with the original comment attached.

Reply:
Switched to using existing HashFunctions.UtilHash (same implementation for longs).